pkg/isoeditor/rhcos.go

```diff
@@ -28,44 +28,63 @@ func NewEditor(dataDir string) Editor {
 	return &rhcosEditor{workDir: dataDir}
 }
 
-// CreateMinimalISOTemplate Creates the template minimal iso by removing the rootfs and adding the url
-func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, minimalISOPath string) error {
-	extractDir, err := os.MkdirTemp(e.workDir, "isoutil")
-	if err != nil {
-		return err
-	}
-
-	if err = Extract(fullISOPath, extractDir); err != nil {
-		return err
-	}
-
-	if err = os.Remove(filepath.Join(extractDir, "images/pxeboot/rootfs.img")); err != nil {
+func createMinimalISOFromDir(minimalISOPath, extractDir, volumeID, rootFSURL, arch string) error {
+	if err := os.Remove(filepath.Join(extractDir, "images/pxeboot/rootfs.img")); err != nil {
 		return err
 	}
 
-	if err = embedInitrdPlaceholders(extractDir); err != nil {
+	if err := embedInitrdPlaceholders(extractDir); err != nil {
 		log.WithError(err).Warnf("Failed to embed initrd placeholders")
 		return err
 	}
 
-	if err = fixGrubConfig(rootFSURL, extractDir); err != nil {
+	if err := fixGrubConfig(rootFSURL, extractDir); err != nil {
 		log.WithError(err).Warnf("Failed to edit grub config")
 		return err
 	}
+
 	// ignore isolinux.cfg for ppc64le because it doesn't exist
 	if arch != "ppc64le" {
-		if err = fixIsolinuxConfig(rootFSURL, extractDir); err != nil {
+		if err := fixIsolinuxConfig(rootFSURL, extractDir); err != nil {
 			log.WithError(err).Warnf("Failed to edit isolinux config")
 			return err
 		}
 	}
 
+	if err := Create(minimalISOPath, extractDir, volumeID); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateMinimalISOTemplate Creates the template minimal iso by removing the rootfs and adding the url
+func (e *rhcosEditor) CreateMinimalISOTemplate(fullISOPath, rootFSURL, arch, minimalISOPath string) error {
+	extractDir, err := os.MkdirTemp(e.workDir, "isoutil")
+	if err != nil {
+		return err
+	}
+
+	if err = Extract(fullISOPath, extractDir); err != nil {
+		return err
+	}
+
 	volumeID, err := VolumeIdentifier(fullISOPath)
 	if err != nil {
 		return err
 	}
 
-	if err = Create(minimalISOPath, extractDir, volumeID); err != nil {
+	err = createMinimalISOFromDir(minimalISOPath, extractDir, volumeID, rootFSURL, arch)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateMinimalISO Creates the minimal iso by removing the rootfs and adding the url for agent installer
+func CreateMinimalISO(minimalISOPath, extractDir, volumeID, rootFSURL, arch string) error {
+	err := createMinimalISOFromDir(minimalISOPath, extractDir, volumeID, rootFSURL, arch)
```

Comment:
This can further be changed to `CreateMinimalISOFromDir` i.e. exported and agent installer can call it.

Reply:
TBH defining another function just to have a different name exported externally vs internally seems a bit unnecessary.
Perhaps you could rename `createMinimalISOFromDir` to e.g. `AssembleMinimalISO` or somthing.

Reply:
Ok, so renamed the func. The original `CreateMinimalISOTemplate()` now calls `CreateMinimalISO()`. The actual minimal ISO is "created" in `CreateMinimalISO()` and the same will be called by agent installer